securesignals-gma-devapp/src/main/java/com/uid2/dev/MyActivity.java

```diff
@@ -0,0 +1,155 @@
+package com.uid2.dev;
+
+import static com.google.ads.AdRequest.LOGTAG;
+
+import android.os.Bundle;
+import androidx.appcompat.app.AppCompatActivity;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import com.google.android.gms.ads.AdRequest;
+import com.google.android.gms.ads.AdView;
+import com.google.android.gms.ads.MobileAds;
+import com.google.android.gms.ads.RequestConfiguration;
+import com.google.android.gms.ads.initialization.InitializationStatus;
+import com.google.android.gms.ads.initialization.OnInitializationCompleteListener;
+import com.uid2.UID2Manager;
+import com.uid2.data.UID2Identity;
+import com.uid2.devapp.R;
+
+import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+
+/**
+ * Port of <a href="https://github.com/googleads/googleads-mobile-android-examples/tree/main/java/admob/BannerExample">BannerExample</a>
+ */
+public class MyActivity extends AppCompatActivity {
```

Comment:
Similar to the IMA review, there are a number of issues in this file, including:
 - `UID2Manager` should be initialised in an `Application` class, not an Activity
 - Line lengths of comments should use full 120
 - Deserialisation of initial identity could be improved (as in, using the existing `fromJson` method - once made static)
 - Name of activity layout should be `main_activity`.
 - etc

Reply:
* `UID2Manager` initialization moved to `GMADevApplication`.
* Changed `MyActivity` to `BannerActivity` to differentiate between DevApps.  Primary DevApp already uses `MainActivity`